app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/settings/SettingsFragment.kt

```diff
@@ -1,11 +1,102 @@
 package com.github.braillesystems.learnbraille.ui.screens.settings
 
+import android.app.AlertDialog
 import android.os.Bundle
+import android.text.InputType
+import android.widget.EditText
+import androidx.lifecycle.lifecycleScope
+import androidx.preference.Preference
+import androidx.preference.SwitchPreferenceCompat
+import com.github.braillesystems.learnbraille.COURSE
 import com.github.braillesystems.learnbraille.R
+import com.github.braillesystems.learnbraille.data.entities.CurrentStep
+import com.github.braillesystems.learnbraille.data.repository.MutableTheoryRepository
+import com.github.braillesystems.learnbraille.data.repository.PreferenceRepository
+import com.github.braillesystems.learnbraille.utils.contextNotNull
+import com.github.braillesystems.learnbraille.utils.preferences
+import com.github.braillesystems.learnbraille.utils.toast
+import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
+import timber.log.Timber
 
 class SettingsFragment : androidx.preference.PreferenceFragmentCompat() {
 
+    private val preferenceRepository: PreferenceRepository by inject()
+    private val theoryRepository: MutableTheoryRepository by inject()
+
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.settings_hierarchy, rootKey)
+
+        val preference = findPreference<Preference>(
+            getString(R.string.preference_teacher_mode_enabled)
+        ) ?: error("No teacher mode preference found")
+
+        preference.onPreferenceChangeListener =
+            Preference.OnPreferenceChangeListener { _, newValue ->
+                require(preference is SwitchPreferenceCompat)
+                require(newValue is Boolean)
+
+                val saveCurrStepPreference =
+                    "save_curr_step_teacher_mode_${preferenceRepository.currentUserId}"
+                if (newValue) {
+                    askCode {
+                        val code = 8436
```

Comment:
Would you mind replacing it with a more memorable `0001`?

Reply:
It should not be so memorable for students, I guess :)

Reply:
As you wish, let it be `8436`